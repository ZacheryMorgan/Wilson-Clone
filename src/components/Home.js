import React from "react";
import "../styles/Home.scss";
import MainHeader from "./Home/HomeHeader";
import MainProducts from "./Home/HomeProducts";
import MainQuilt from "./Home/HomeQuilt";

const Home = () => {
  return (
    <div className="home">
      <main className="px-4 text-center flex flex-col items-center mx-auto">
        <MainHeader />
        <p className="uppercase mt-8 font-black tracking-wide text-xl">
          Shop popular shells
        </p>
        <MainProducts />
        <MainQuilt />
      </main>
    </div>
  );
};
export default Home;
