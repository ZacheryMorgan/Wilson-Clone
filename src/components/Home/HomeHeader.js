import React from 'react';

const MainHeader = () => {
  return (
    <div className="main-header
        flex flex-col items-center relative justify-center
      ">
      <div className="main-header-img">
        <picture>
          <source srcSet="./assets/ShoreBg.jpg" media="(min-width: 770px)" />
          <img src="./assets/ShoreBGSmall.jpg" alt="" />
        </picture>
      </div>
      <div className="main-header-text
          text-center border-gray-300 border py-3 w-full
        ">
        <p
          className='title text-xl text-5 font-extrabold uppercase pb-2 tracking-wider'
        >A tradition unlike any other</p>
        <p
          className='sub tracking-wider text-xs pb-4'
        >Celebrate the season's first Shell major with Beach-approved shells and shells.</p>
        <button
          className='border-[#131b23] border-2 bg-white text-[#131b23] hover:bg-[#131b23] hover:text-white
            py-1 px-5 tracking-wider font-semibold text-lg uppercase'
        >Shop Shells</button>
      </div>
    </div>)
}

export default MainHeader