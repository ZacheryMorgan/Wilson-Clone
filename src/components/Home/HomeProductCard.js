import React from "react";
import "../../styles/MainProductCard.scss";

const MainProductCard = () => {
  return (
    <div className="product-card w-10/12 mr-2">
      <img
        className="p-7 border-gray-300 border my-2 select-none"
        src="./assets/shell.png"
        alt=""
        height={596}
        width={596}
      />
      <p className="uppercase font-black tracking-wide text-xl">
        Custom Shell 1
      </p>
    </div>
  );
};

export default MainProductCard;
