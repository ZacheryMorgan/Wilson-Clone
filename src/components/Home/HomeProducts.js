import React from "react";
import MainProductCard from "./HomeProductCard";
import { FaChevronLeft, FaChevronRight } from "react-icons/fa";
import "../../styles/MainProduct.scss";

const MainProducts = () => {
  let width;
  window.addEventListener("resize", () => {
    const cardWrapper = document.querySelector(".card-wrapper")
    cardWrapper.style.transform = "translate3d(0px, 0px, 0px)"
    document.querySelector(".left").style.fill = "#d1d5db";
    document.querySelector(".right").style.fill = "#101820";

    // if (window.innerWidth >= 771) {
    //   return;
    // }
    width = JSON.stringify(document.querySelector(".product-card").offsetWidth);
    // cardWrapper.style.transform = `translate3d(${Math.floor(
    //   -width - 8
    // )}px, 0px, 0px)`;
    return width;

  });

  window.addEventListener("load", () => {
    width = JSON.stringify(document.querySelector(".product-card").offsetWidth);
    return width;
  });

  const scrollRight = () => {
    document.querySelector(".left").style.fill = "black";
    const cardWrapper = document.querySelector(".card-wrapper");
    let pos = cardWrapper.style.transform;
    if (!pos || pos == "translate3d(0px, 0px, 0px)") {
      cardWrapper.style.transform = `translate3d(${Math.floor(
        -width - 8
      )}px, 0px, 0px)`;
    } else if (pos == `translate3d(${Math.floor(-width - 8)}px, 0px, 0px)`) {
      cardWrapper.style.transform = `translate3d(${Math.floor(
        -width * 2 - 8
      )}px, 0px, 0px)`;
    } else if (
      pos == `translate3d(${Math.floor(-width * 2 - 8)}px, 0px, 0px)`
    ) {
      document.querySelector(".right").style.fill = "#d1d5db";
      cardWrapper.style.transform = `translate3d(${Math.floor(
        -width * 3 - 8
      )}px, 0px, 0px)`;
    }
  };
  const scrollLeft = () => {
    document.querySelector(".right").style.fill = "black";
    const cardWrapper = document.querySelector(".card-wrapper");
    let pos = cardWrapper.style.transform;
    if (!pos) {
      return;
    } else if (pos == `translate3d(${Math.floor(-width - 8)}px, 0px, 0px)`) {
      document.querySelector(".left").style.fill = "#d1d5db";
      cardWrapper.style.transform = `translate3d(0px, 0px, 0px)`;
    } else if (
      pos == `translate3d(${Math.floor(-width * 2 - 8)}px, 0px, 0px)`
    ) {
      cardWrapper.style.transform = `translate3d(${Math.floor(
        -width - 8
      )}px, 0px, 0px)`;
    } else if (
      pos == `translate3d(${Math.floor(-width * 3 - 8)}px, 0px, 0px)`
    ) {
      cardWrapper.style.transform = `translate3d(${Math.floor(
        -width * 2 - 8
      )}px, 0px, 0px)`;
    }
  };

  return (
    <div className="container relative overflow-hidden">
      <FaChevronLeft
        onClick={scrollLeft}
        className="left text-black w-14 h-14 absolute left-0 bottom-2/4 cursor-pointer z-10 fill-gray-300"
        width="0"
      />
      <FaChevronRight
        onClick={scrollRight}
        className="right text-black w-14 h-14 absolute right-0 bottom-2/4 cursor-pointer z-10"
        width="0"
      />
      <div className="card-wrapper flex w-auto transition-all duration-500 ease-out">
        <MainProductCard />
        <MainProductCard />
        <MainProductCard />
        <MainProductCard />
      </div>
    </div>
  );
};

export default MainProducts;
