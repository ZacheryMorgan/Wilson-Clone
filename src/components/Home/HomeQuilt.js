import React, { useEffect } from "react";
import MainQuiltItem from "./HomeQuiltItem";

const MainQuilt = () => {

  useEffect(() => {
    const quiltItem = document.querySelectorAll('.quilt-item')

    quiltItem.forEach(item => {
      observer.observe(item)
    })

  })

  const quiltItemOptions = {
    root: null,
    threshold: .5,
    rootMargin: '0px',
  }

  const observer = new IntersectionObserver((entries, observer) => {

    entries.forEach(entry => {
      if (entry.isIntersecting) {
        entry.target.classList.add('appear')
        observer.unobserve(entry.target)
      }
    })

  }, quiltItemOptions)


  return (
    <div className="quilt-wrapper w-full mt-8 gap-2 grid mb-4">
      <MainQuiltItem />
      <MainQuiltItem />
      <MainQuiltItem />
      <MainQuiltItem />
    </div>
  );
};

export default MainQuilt;
