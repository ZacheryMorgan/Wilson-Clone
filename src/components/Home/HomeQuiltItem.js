import React from "react";
import "../../styles/MainQuiltItem.scss";

const MainQuiltItem = () => {
  return (
    <div className="relative">
      <img src="./assets/beach710.jpg" alt="" className="w-full" />
      <div className="quilt-content text-left absolute bottom-0 left-0 p-3">
        <p className="text-lg font-black text-white pb-3">SAVE ON SETS</p>
        <p className="text-sm text-white pb-3">
          Save when you buy a <br /> performance top and bottom together
        </p>
        <button
          className="border-[#131b23] border-2 bg-white text-[#131b23] hover:bg-[#131b23] hover:text-white
            py-1 px-5 tracking-wider font-semibold text-base uppercase"
        >
          MEN'S SETS
        </button>
      </div>
    </div>
  );
};

export default MainQuiltItem;
