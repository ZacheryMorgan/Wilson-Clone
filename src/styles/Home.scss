* {
  color: #101820;
}

@media only screen and (min-width: 770px) {
  main {
    .main-header {
      .main-header-text {
        position: absolute;
        border: unset;

        .title {
          font-size: 1.5rem;
          line-height: 1;
          padding-bottom: 12px;
          font-weight: bold;
        }

        .sub {
          font-size: 0.8rem;
          line-height: 1;
          letter-spacing: 1px;
          font-weight: bold;
          padding-bottom: 25px;
        }

        button {
          padding: 12px 20px 12px 20px;
          font-size: 1rem;
          line-height: 1;
          font-weight: bold;
        }
      }
    }
  }
}

// .title text-5xl font-extrabold uppercase pb-2
// .sub text-xl font-semibold pb-4
// button py-3 px-6 text-2xl font-semibold
