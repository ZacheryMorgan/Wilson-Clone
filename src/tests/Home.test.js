import React from "react";
import { render } from "@testing-library/react";
import Home from '../components/Home';

describe('Home component', () => {
  it('renders Home component correctly', () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot()
  })
})