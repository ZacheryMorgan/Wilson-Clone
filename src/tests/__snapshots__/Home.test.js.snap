// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders Home component correctly 1`] = `
<div>
  <main
    class="flex flex-col p-40 pt-80 pl-60"
  >
    <h1
      class="header text-6xl font-bold text-white pb-2"
    >
      C:Shell
    </h1>
    <h3
      class="description text-3xl text-white"
    >
      Bring the ocean home.
    </h3>
  </main>
</div>
`;
